pkg/dns/dns.go

```diff
@@ -433,6 +464,28 @@ func (kd *KubeDNS) generateRecordsForHeadlessService(e *v1.Endpoints, svc *v1.Se
 	return nil
 }
 
+func (kd *KubeDNS) isHeadlessServiceNamed(svc *v1.Service, podReference *v1.ObjectReference) bool {
+	if podReference != nil {
+		pod, err := kd.kubeClient.Core().Pods(podReference.Namespace).Get(podReference.Name)
```

Comment:
We should not be accessing API for pods in DNS.  We Should have enough information in the Endpoints structure - if the EndpointAddress has a .Hostname set *and* the Service is headless, we should just take it, I think.  It's a little sloppy, but I am worried about the churn of doing this for every backend of every headless service.

Maybe we can pre-filter by the existence of EndpointAddress.Hostname, so we only need to do this on those endpoints that elect themselves to have a hostname.  Still seems like a DoS vector.  @smarterclayton  @bowei